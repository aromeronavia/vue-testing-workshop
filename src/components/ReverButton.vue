<template>
  <button
    :class="{
      'btn-primary': type === 'primary',
      'btn-warning': type === 'warning',
      'btn-danger': type === 'danger',
    }"
  >
    <slot></slot>
  </button>
</template>

<script>
export default {
  name: 'ReverButton',
  props: {
    type: {
      type: String,
      validator: function (value) {
        return ['primary', 'warning', 'danger'].indexOf(value) !== -1;
      },
      required: false,
      default: 'primary',
    },
  },
};
</script>

<style>
  .btn-primary {
    background-color: blue;
  }

  .btn-warning {
    background-color: yellow;
  }

  .btn-danger {
    background-color: red;
  }
</style>
