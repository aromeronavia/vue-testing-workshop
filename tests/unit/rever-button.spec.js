import { shallowMount } from '@vue/test-utils';
import ReverButton from '@/components/ReverButton.vue';
import ButtonPageObject from '../pageObjects/ReverButton.po';

describe('ReverButton.vue', () => {
  describe('Button', () => {
    const mountButton = (options = {}) => {
      const wrapper = shallowMount(ReverButton, {
        propsData: {
          onClick: options.onClickStub || jest.fn(),
          type: options.type,
        },
      });
      return new ButtonPageObject(wrapper);
    };

    it('renders as primary button', () => {
      const button = mountButton({ type: 'primary' });

      expect(button.isPrimary()).toBe(true);
    });

    it('renders as warning button', () => {
      const button = mountButton({ type: 'warning' });

      expect(button.isWarning()).toBe(true);
    });

    it('renders as danger button', () => {
      const button = mountButton({ type: 'danger' });

      expect(button.isDanger()).toBe(true);
    });
  });

  describe('Snapshot', () => {
    it('should render content correctly', () => {
      const wrapper = shallowMount(ReverButton, {
        propsData: {
          onClick: jest.fn(),
        },
      });

      expect(wrapper.vm.$el).toMatchSnapshot();
    });
  });
});
