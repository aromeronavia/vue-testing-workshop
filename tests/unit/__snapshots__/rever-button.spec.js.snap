// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReverButton.vue Snapshot should render content correctly 1`] = `
<button
  class="btn-primary"
/>
`;
